pyro/infer/trace_elbo.py

```diff
@@ -77,7 +77,7 @@ def loss(self, model, guide, *args, **kwargs):
         """
         elbo = 0.0
         for model_trace, guide_trace in self._get_traces(model, guide, *args, **kwargs):
-            elbo_particle = (model_trace.log_prob_sum() - guide_trace.log_prob_sum()).item()
+            elbo_particle = torch_item(model_trace.log_prob_sum()) - torch_item(guide_trace.log_prob_sum())
```

Comment:
why did torch_item become necessary?

Reply:
To increase precision in light of the reordering of operations in this PR, I've switched from accumulating `elbo_particle` as a (typically 32bit) `FloatTensor` to accumulating it as a 64-bit Python float. To achieve this, we could thus use `.item()` except that in cases of empty models or guides, `.log_prob_sum()` is a float rather than a tensor. Hence the special handling `x if isinstance(x, Number) else x.item()`.